packages/node-cli/src/commands/actionresult/ActionResult.js

```diff
@@ -0,0 +1,42 @@
+/*
+** Copyright (c) 2019 Oracle and/or its affiliates.  All rights reserved.
+** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+*/
+'use strict';
+const assert = require('assert');
+const ActionResultStatus = require('../actionresult/ActionResultStatus');
+
+module.exports = class ActionResult {
+
+	constructor(options) {
+		assert(options);
+		assert(options.status, "status is required when creating an ActionResult object.");
+
+		if (options.status === ActionResultStatus.SUCCESS) {
+			assert(options.context, "context is required when ActionResult is a success.");
```

Comment:
What is context??

Reply:
Deleted context and added _data, _resultMessages variables to ActionResult,
Created sons of ActionResult for special commands like CreateProject or Deploy